### PR TITLE
web/console: casper queue

### PR DESCRIPTION
**Pull request by AnnaTsybel**

### CODE REVIEW

**Review by vladdezhniuk — CHANGES_REQUESTED** [review]
Split backend and frontend in separate prs @AnnaTsybel  @APrisiazhniuk

File: web/console/src/app/views/Match/MatchScore/index.tsx [hunk 1]
```diff
@@ -1,31 +1,44 @@
...
+const CASPER_WALLET_TYPE = 'casper_wallet_address';
+const METAMASK_WALLET_TYPE = 'wallet_address';
+
 export const MatchScore: React.FC = () => {
+    const dispatch = useDispatch();
+
     const [queueClient, setQueueClient] = useState<QueueClient | null>(null);
 
     const onboarding = useMemo(() => new MetaMaskOnboarding(), []);
+
```
Comment (TarasPotsikaylo) [inline web/console/src/app/views/Match/MatchScore/index.tsx:31]: no extra space between constants

File: web/console/src/app/views/Match/MatchScore/index.tsx [hunk 2]
```diff
@@ -34,8 +47,20 @@ export const MatchScore: React.FC = () => {
     /** Variable describes that it needs alllow to add address or forbid add adress. */
     const CONFIRM_ADD_WALLET: string = 'do you allow us to take your address?';
 
+    /** sets user info */
+    async function setUser() {
```
Comment (TarasPotsikaylo) [inline web/console/src/app/views/Match/MatchScore/index.tsx:51]: 51 - why function declaration ?
63 - expression function ? 
need only one

File: web/console/src/app/views/Match/MatchScore/index.tsx [hunk 3]
```diff
@@ -70,13 +95,105 @@ export const MatchScore: React.FC = () => {
         }
     };
 
+    /** Adds casper wallet address for earning reward. */
+    const addCasperWallet = () => {
+        try {
+            const currentQueueClient = getCurrentQueueClient();
+
```
Comment (TarasPotsikaylo) [inline web/console/src/app/views/Match/MatchScore/index.tsx:102]: delete extra spaces

File: web/console/src/app/views/Match/MatchScore/index.tsx [hunk 4]
```diff
@@ -70,13 +95,105 @@ export const MatchScore: React.FC = () => {
...
+                },
+                {
+                    walletType: CASPER_WALLET_TYPE,
+                    mint: casperMint,
+                },
+                {
+                    walletType: METAMASK_WALLET_TYPE,
+                    mint: metamaskMint,
+                },
+            ];
```
Comment (TarasPotsikaylo) [inline web/console/src/app/views/Match/MatchScore/index.tsx:184]: 123-136 - duplicate logic

File: web/console/src/queue/service.ts [hunk 5]
```diff
@@ -17,6 +17,10 @@ export class QueueService {
         this.queueClient.actionAllowAddress(wallet, nonce);
     };
 
+    /** Sends action that indicates that the client allows to add address of wallet. */
+    public casperActionAllowAddress(wallet: string, walletType: string, squadId:string): void {
```
Comment (TarasPotsikaylo) [inline web/console/src/queue/service.ts:21]: linter

File: web/console/src/queue/service.ts [hunk 6]
```diff
@@ -53,6 +57,11 @@ export const queueActionAllowAddress = (wallet: string, nonce: number) => {
     queueService.actionAllowAddress(wallet, nonce);
 };
 
+/** Sends action that indicates that the client allows to add address of wallet. */
+export const queueCasperActionAllowAddress = (wallet: string, walletType: string, squadId:string) => {
```
Comment (TarasPotsikaylo) [inline web/console/src/queue/service.ts:61]: linter

File: web/console/src/api/queue.ts [hunk 7]
```diff
@@ -24,6 +24,14 @@ export class QueueClient {
         this.ws.send(JSON.stringify({ action, WalletAddress, nonce }));
     };
 
+    /** Sends action that indicates that the client allows to add address of wallet. */
+    public casperActionAllowAddress(casperWallet: string, walletType: string, squadId:string) {
+        const action: string = 'allowAddress';
```
Comment (OlehPerehuda) [inline web/console/src/api/queue.ts:29]: Better to create enum for all available actions

File: web/console/src/app/views/Match/MatchScore/index.tsx [hunk 8]
```diff
@@ -1,31 +1,44 @@
...
+import { setCurrentUser } from '@/app/store/actions/users';
+import CasperTransactionService from '@/casper';
+
+import coin from '@static/img/match/money.svg';
 
 import './index.scss';
 
+const VELAS_WALLET_TYPE = 'velas_wallet_address';
+const CASPER_WALLET_TYPE = 'casper_wallet_address';
+const METAMASK_WALLET_TYPE = 'wallet_address';
```
Comment (OlehPerehuda) [inline web/console/src/app/views/Match/MatchScore/index.tsx:23]: not critical but also these better to implement with enum

File: web/console/src/app/views/Match/MatchScore/index.tsx [hunk 9]
```diff
@@ -34,8 +47,20 @@ export const MatchScore: React.FC = () => {
...
+    /** sets user info */
+    async function setUser() {
+        try {
+            await dispatch(setCurrentUser());
+        } catch (error: any) {
+            toast.error('Something went wrong', {
+                position: toast.POSITION.TOP_RIGHT,
+                theme: 'colored',
+            });
+        }
```
Comment (OlehPerehuda) [inline web/console/src/app/views/Match/MatchScore/index.tsx:59]: just for now it is ok.
But in future we could create notifications configs.
We will hold all possible error messages and methods
something like this.
https://github.com/BoostyLabs/amuzed/blob/master/web/console/src/plugins/toastNotifications.ts
https://github.com/BoostyLabs/amuzed/blob/master/web/console/src/app/configs/toastNoficationsMessages.json
But toast sufix was not good idea :)

File: web/console/src/casper/index.ts [hunk 10]
```diff
@@ -121,6 +121,33 @@ class CasperTransactionService {
...
     }
+
+    /** Mints a token */
+    async mintUDT(transaction: CasperMatchTransaction, rpcNodeAddress: string): Promise<void> {
+        try {
+            const runtimeArgs = RuntimeArgs.fromMap({
+                'signature': CLValueBuilder.string(transaction.signature),
+                'value': CLValueBuilder.u256(transaction.value),
+                'nonce': CLValueBuilder.u64(transaction.nonce),
+            });
```
Comment (OlehPerehuda) [inline web/console/src/casper/index.ts:132]: i though we already have enum for runtime args

File: web/console/src/casper/index.ts [hunk 11]
```diff
@@ -121,6 +121,33 @@ class CasperTransactionService {
...
+
+            const isConnected = window.casperlabsHelper.isConnected();
+
+            if (!isConnected) {
+                await window.casperlabsHelper.requestConnection();
+            }
+
+            const signature = await this.contractSign('claim', runtimeArgs, this.paymentAmount, transaction.casperTokenContract.address);
+
+            await this.client.claim(rpcNodeAddress, JSON.stringify(signature), this.walletAddress);
```
Comment (OlehPerehuda) [inline web/console/src/casper/index.ts:142]: is it better to place these logic on service/api level?

File: web/console/src/app/views/Match/MatchScore/index.tsx (hunk 4 above)
Comment (AnnaTsybel) [reply web/console/src/app/views/Match/MatchScore/index.tsx:184]: I will create service for this in another pr

**Pull request merged**